lti_consumer/migrations/0012_courseeditltifieldsenabledflag.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 2.2.20 on 2021-05-05 11:56
+import logging
+
+import django.db.models.deletion
+import opaque_keys.edx.django.models
+from django.conf import settings
+from django.db import migrations, models
+
+log = logging.getLogger(__name__)
+
+
+class CreateModelIfNotExists(migrations.CreateModel):
+    """
+    Creates the database table if it doesn't already exist.
+
+    This can be used to move a database model from one app to another.
+    """
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        # Get the name of the database table
+        db_table = to_state.apps.get_model(app_label, self.name)._meta.db_table
+        # If the database table for this model already exists, do nothing, otherwise
+        # create the table as usual.
+        if db_table in schema_editor.connection.introspection.table_names():
+            log.info(f"{db_table} already exists. Skipping creation.")
+        else:
+            super().database_forwards(app_label, schema_editor, from_state, to_state)
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        # Do nothing when applying this is reverse because the original creation of this
+        # table was handled by edx-platform, so we don't want to delete the table on reversal.
+        pass
```

Comment:
This customisation is needed because we're moving the model from edx-platform to here, but we can't have a dependency between the migrations. For a completely new instance it's possible the table might not exist, and we'd have to create it,  However for an existing instance, the table would already exist, and Django would throw and error running the migration since the table already exists. 

So here we check if the table exists, and if it does, we simply skip creating the table _in the database_ while still creating it _in the schema_ (i.e. we are not overriding schema_forwards)..